src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -85,6 +85,10 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
                     if (state === ZipModalEvent.CheckoutReferred && checkoutId) {
                         await this._prepareForReferredRegistration(payment.methodId, checkoutId);
 
+                        if (this._paymentMethod?.initializationData?.experimentZipDeferredFlow) {
```

Comment:
Take a glance at this comment https://github.com/bigcommerce/bigcommerce/pull/39599#discussion_r581046249